lib/automatic_namespaces/autoloader.rb

```diff
@@ -20,16 +22,16 @@ def set_namespace_for(pack_dir, package_namespace)
     Rails.application.config.watchable_dirs[pack_dir] = [:rb]
   end
 
-  def pack_directories(pack_root_dir)
-    Dir.glob("#{pack_root_dir}/**/app/*").reject { |dir| non_namspaced_directory(dir) }
+  def pack_directories(pack_root_dir, metadata)
+    Dir.glob("#{pack_root_dir}/**/app/*").select { |dir| namespaced_directory?(dir, metadata) }
   end
 
-  def non_namspaced_directory(dir)
-    dir.include?('/app/assets') ||
-      dir.include?('/app/helpers') || # Rails assumes helpers are global, not namespaced
-      dir.include?('/app/inputs') || # Not sure how to namespace form inputs
-      dir.include?('/app/javascript') ||
-      dir.include?('/app/views')
+  def namespaced_directory?(dir, metadata)
+    excluded_directories(metadata).none? { |excluded_dir| dir.include?(excluded_dir) }
+  end
+
+  def excluded_directories(metadata)
```

Comment:
I see the problem.. thanks for addressing the logic error. It appears that it stems from a naming confusion, in that these are all called 'dirs' but really, some are relatively named paths and some are absolutely named paths. We could probably improve the code quite a bit by naming things better, and possibly even harmonizing to use the same kind of thing in both cases.

Reply:
> We could probably improve the code quite a bit by naming things better, and possibly even harmonizing to use the same kind of thing in both cases.

Totally agree, and thank you again for taking the time to review my changes. Is this something you'd like me to take a pass at before we merge this PR in?

Reply:
Let's go with this for now.

Reply:
> Let's go with this for now.

Awesome. Let me know if there is anything I can do to get this merged and released. We've got a few teams hoping to leverage this feature soon.